tests/net/socket/socketpair/src/test_socketpair_happy_path.c

```diff
@@ -115,10 +115,12 @@ static void happy_path(
 
 		memset(actual_msg, 0, sizeof(actual_msg));
 
+		len = 0;
 		res = recvfrom(sv[(!i) & 1], actual_msg, sizeof(actual_msg), 0,
 			NULL, &len);
 
 		zassert_not_equal(res, -1, "recvfrom(2) failed: %d", errno);
+		zassert_true(res >= 0, "recvfrom must return >= -1: %d", res);
```

Comment:
```suggestion
		zassert_true(res >= 0, "recvfrom must return >= 0: %d", res);
```

Reply:
Technically it can return -1 if there is an error. It should never return -2 etc though

Reply:
I thought you put `res >= 0` because you have `zassert_not_equal(res, -1, ...)` right above i.t